sw/device/lib/dif/dif_pwrmgr.h

```diff
@@ -0,0 +1,497 @@
+  /**
+   * Whether the core clock should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t core_clock;
+  /**
+   * Whether the input/output (IO) clock should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t io_clock;
+  /**
+   * Whether the main power domain should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t main_power_domain;
+} dif_pwrmgr_low_power_config_t;
+
+/**
+ * Configure low power state.
+ *
+ * @param pwrmgr A power manager.
+ * @param new_config New low power state configuration.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_config_set(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_low_power_config_t *new_config);
+
+/**
+ * Get the current low power state configuration.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_config Current low power state configuration.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_config_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_low_power_config_t *cur_config);
+
+/**
+ * Request types.
+ */
+typedef enum dif_pwrmgr_req_type {
+  /**
+   * A wake-up request.
+   */
+  kDifPwrmgrReqTypeWakeUp,
+  /**
+   * A reset request.
+   */
+  kDifPwrmgrReqTypeReset,
+} dif_pwrmgr_req_type_t;
+
+/**
+ * Lock sources for a request type.
+ *
+ * Once sources for a request type is locked, they cannot be changed
+ * until the chip is reset.
+ *
+ * @param pwrmgr A power manager.
+ * @param req_type A request type.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_request_sources_lock(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_req_type_t req_type);
+
+/**
+ * Check if sources for a request type is locked.
+ *
+ * @param pwrmgr A power manager.
+ * @param req_type A request type.
+ * @param is_locked Whether sources for the request type is locked.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_request_sources_is_locked(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_req_type_t req_type,
+    dif_pwrmgr_toggle_t *is_locked);
+
+/**
+ * Return codes for functions that write to configuration registers.
+ */
+typedef enum dif_pwrmgr_config_result {
+  /**
+   * The call succeeded.
+   */
+  kDifPwrmgrConfigResultOK = 0,
+  /**
+   * A non-specific error occurred and the hardware is in an invalid or
+   * irrecoverable state.
+   */
+  kDifPwrmgrConfigResultError = 1,
+  /**
+   * The caller supplied invalid arguments but the call did not cause any
+   * side-effects and the hardware is in a valid and recoverable state.
+   */
+  kDifPwrmgrConfigResultBadArg = 2,
+  /**
+   * The register that the function needs to write to is locked.
+   */
+  kDifPwrMgrConfigResultLocked,
+} dif_pwrmgr_config_result_t;
+
+/**
+ * Enable or disable sources for a request type.
+ *
+ * Setting `new_sources[i]` to 1 enables source `i` for `req_type` requests.
```

Comment:
This is somewhat confusing as `i` refers to a bit index in `new_sources`.

Reply:
Updated comment.